### PR TITLE
Add selinux base policy for hirte and hirte-agent

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

File: selinux/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+TARGETS?=hirte
```
Comment (mwperina) [inline selinux/Makefile:1]: This is not going to work for RPM installations, we are using meson to build and install, Makefile is available just as a facade for developers and make should call meson to perform stuff (still not true for some invocations like lint or codespell, but we will get there).
So could you please convert it to meson and adapt spec file accordingly?

Comment (rhatdan) [reply selinux/Makefile:1]: Will adapt spec when this is ready. Doing meson will take longer since I don't know it.

Comment (rhatdan) [reply selinux/Makefile:1]: But we will need to exec makefile to build selinux policy, I think.

Comment (mwperina) [reply selinux/Makefile:1]: > But we will need to exec makefile to build selinux policy, I think.

If you want I can take a look how to do it in meson

Comment (rhatdan) [reply selinux/Makefile:1]: Sure, but the compile of selinux is using the /usr/share/selinux/devel/Makefile provided by the selinux-policy-devel package.

Comment (rhatdan) [reply selinux/Makefile:1]: The big part that is missing is to integrate this build and install into meson.

Comment (mwperina) [reply selinux/Makefile:1]: Should be fixed now

File: selinux/hirte_agent_selinux.8
Comment (mwperina) [inline selinux/hirte_agent_selinux.8:1]: Wouldn't it be better to align with other existing man pages and rewrite them using Makrdown?

Comment (rhatdan) [reply selinux/hirte_agent_selinux.8:1]: These man mages are generated based on the selinux-policy on the system. I did not write them.

**Comment by rhatdan** [comment]
@mwperina Changes LGTM
Lets get this merged and then we can continue to evolve it.

**Review by mwperina — APPROVED** [review]
+1

File: selinux/hirte_agent_selinux.8
Comment (mwperina) [reply selinux/hirte_agent_selinux.8:1]: Ah, OK

File: hirte.spec.in [hunk 2]
```diff
@@ -91,6 +92,38 @@ This package contains the node agent.
 %{_unitdir}/hirte-dep@.service
 %{_userunitdir}/hirte-dep@.service
 
+%package selinux
+Summary:  Hirte SELinux policy
+BuildRequires: checkpolicy
+BuildRequires: selinux-policy-devel
+
+%if "%{_selinux_policy_version}" != ""
+Requires: selinux-policy >= %{_selinux_policy_version}
```
Comment (pypingou) [inline hirte.spec.in:101]: Should we `Requires: selinux-policy` if not specific version is specified? (ie: in a else clause?)

**Comment by alexlarsson** [comment]
lgtm

**Review by pypingou — APPROVED** [review]
one question, but otherwise looks good to me

File: hirte.spec.in (hunk 2 above)
Comment (mwperina) [reply hirte.spec.in:101]: AFAIK you need to require the exact selinux-policy with which you built it. In theory even the newer one should work, but we already had a case in oVirt where this didn't work.

Anyway the above is the official recommendation:
https://fedoraproject.org/wiki/SELinux_Policy_Modules_Packaging_Draft#Runtime_Dependencies

Comment (pypingou) [reply hirte.spec.in:101]: ah ok, so the macro is prepopulated by `selinux-policy` cool, thanks for the info!

**Pull request merged**